projects/plugins/jetpack/modules/masterbar/profile-edit.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * WP-Admin Profile edit.
+ *
+ * @package Jetpack
+ */
+
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+
+/**
+ * Hides profile fields for WordPress.com connected users.
+ *
+ * @param WP_User $user The current WP_User object.
+ */
+function masterbar_hide_profile_fields( $user ) {
+	$connection_manager = new Connection_Manager( 'jetpack' );
+	if ( ! $connection_manager->is_user_connected( $user->ID ) ) {
+		// If this is a local user, show the default UX.
+		return;
+	}
+	// Since there is no hook for altering profile fields, we will use CSS and JS.
+	$name_info_wpcom_link_message = sprintf(
+		__( 'WordPress.com users can change their profile\\\'s basic details ( i.e., First Name, Last Name, Display Name, About ) in <a href="%1$s">WordPress.com Profile settings.</a>', 'jetpack' ),
+		'https://wordpress.com/me',
+	);
+	$contact_info_wpcom_link_message = sprintf(
+		__( 'WordPress.com users can change their profile\\\'s email & website address in <a href="%1$s">WordPress.com Account settings.</a>', 'jetpack' ),
+		'https://wordpress.com/me/account',
+	);
+	?>
+	<style>
```

Comment:
Does this produce valid HTML? According to [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/style) seems we should avoid inserting the `style` tag in the `body` element:

> The <style> element must be included inside the <head> of the document.

Reply:
You are correct. I was influenced by the `grofiles.php` approach in WPCOM. I have refactored it to use an external stylesheet. 3ef273e